Sources_v3/Workers/MessageUpdater_Tests.swift

```diff
@@ -0,0 +1,559 @@
+            completionCalled = true
+        }
+        
+        // Load the message
+        var message: MessageDTO? {
+            database.viewContext.message(id: createdMessageId)
+        }
+        
+        AssertAsync {
+            // Assert completion is called
+            Assert.willBeTrue(completionCalled)
+            // Assert message still has `pendingSync` state
+            Assert.willBeEqual(message?.localMessageState, .pendingSync)
+            // Assert message text is updated correctly
+            Assert.willBeEqual(message?.text, updatedText)
+        }
+    }
+    
+    func test_editMessage_updatesMessageLocalStateCorrectly_ifLocalStateIsPendingSend() throws {
+        let updatedText: String = .unique
+
+        // Create current user and channel in the database
+        let (_, cid) = try createCurrentUserAndChannel()
+        
+        // Create a new message in the database with `pendingSend` local state
+        let createdMessageId = try createMessage(in: cid, localState: .pendingSend)
+
+        // Edit created message with `updatedText`
+        var completionCalled = false
+        messageUpdater.editMessage(messageId: createdMessageId, text: updatedText) {
+            XCTAssertNil($0)
+            completionCalled = true
+        }
+        
+        // Load the message
+        var message: MessageDTO? {
+            database.viewContext.message(id: createdMessageId)
+        }
+        
+        AssertAsync {
+            // Assert completion is called
+            Assert.willBeTrue(completionCalled)
+            // Assert message still has `pendingSend` state
+            Assert.willBeEqual(message?.localMessageState, .pendingSend)
+            // Assert message text is updated correctly
+            Assert.willBeEqual(message?.text, updatedText)
+        }
+    }
+    
+    func test_editMessage_propogatesError_ifLocalStateIsDeleting() throws {
+        try assert_editMessage_throwsTheError_and_doesNotChangeTheMessage(localState: .deleting)
+    }
+    
+    func test_editMessage_propogatesError_ifLocalStateIsDeletingFailed() throws {
+        try assert_editMessage_throwsTheError_and_doesNotChangeTheMessage(localState: .deletingFailed)
+    }
+    
+    func test_editMessage_propogatesError_ifLocalStateIsSending() throws {
+        try assert_editMessage_throwsTheError_and_doesNotChangeTheMessage(localState: .sending)
+    }
+    
+    func test_editMessage_propogatesError_ifLocalStateIsSendingFailed() throws {
+        try assert_editMessage_throwsTheError_and_doesNotChangeTheMessage(localState: .sendingFailed)
+    }
+    
+    func test_editMessage_propogatesError_ifLocalStateIsSyncing() throws {
+        try assert_editMessage_throwsTheError_and_doesNotChangeTheMessage(localState: .syncing)
+    }
+    
+    func test_editMessage_propogatesError_ifLocalStateIsSyncingFailed() throws {
+        try assert_editMessage_throwsTheError_and_doesNotChangeTheMessage(localState: .syncingFailed)
+    }
```

Comment:
Do you think we need this? Can just make one test like `test_editMessage_returnsError_forUnsupportedLocalStates` and just loop over the unsupported states there? Having this seems like an unnecesary boilerplate.